x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingDeciderService.java

```diff
@@ -813,11 +831,13 @@ Optional<AutoscalingDeciderResult> checkForScaleUp(
             NativeMemoryCapacity updatedCapacity = NativeMemoryCapacity.from(currentScale)
                 .merge(analyticsCapacityAndNewLoad.v1())
                 .merge(anomalyCapacityAndNewLoad.v1())
-                .merge(modelCapacityAndNewLoad.v1())
-                // Since we require new capacity, it COULD be we require a brand new node
-                // We should account for overhead in the tier capacity just in case.
-                .merge(new NativeMemoryCapacity(MachineLearning.NATIVE_EXECUTABLE_CODE_OVERHEAD.getBytes(), 0));
```

Comment:
While it _COULD_ be the case that we require a brand new node, it also _COULD_ be the case that we don't.  Including the overhead here unconditionally was the case of an inconsistency that could lead to scale up/scale down loops.  Fixing this is a nice side effect of changing native capacity to be memory _excluding_ the per-node overhead.